src/stores/reactions.js

```diff
@@ -0,0 +1,240 @@
+			}
+		},
+
+		/**
+		 * Remove a reaction for a given message.
+		 *
+		 * @param {object} payload action payload
+		 * @param {string} payload.token The conversation token
+		 * @param {number} payload.messageId The id of message
+		 * @param {string} payload.reaction The reaction to remove
+		 *
+		 */
+		removeReaction({ token, messageId, reaction }) {
+			Vue.delete(this.reactions[token][messageId], reaction)
+		},
+
+		/**
+		 * Add an actor for a given reaction emoji.
+		 *
+		 * @param {object} payload action payload
+		 * @param {string} payload.token The conversation token
+		 * @param {number} payload.messageId The id of message
+		 * @param {string} payload.reaction The reaction emoji
+		 * @param {object} payload.actor The user who reacted
+		 *
+		 */
+		addActorToReaction({ token, messageId, reaction, actor }) {
+			if (!this.reactions[token]) {
+				Vue.set(this.reactions, token, {})
+
+			}
+			if (!this.reactions[token][messageId]) {
+				Vue.set(this.reactions[token], messageId, {})
+
+			}
+			if (!this.reactions[token][messageId][reaction]) {
+				Vue.set(this.reactions[token][messageId], reaction, [])
+			}
+			const actors = this.reactions[token][messageId][reaction]
+			// Find if actor is already in the list
+			// This is needed when loading as revoking messages fully updates the list
+			const actorExists = actors.find(a => a.actorId === actor.actorId)
+			if (actorExists) {
+				return
+			}
+			actors.push(actor)
+			Vue.set(this.reactions[token][messageId], reaction, actors)
+		},
+
+		/**
+		 * Delete all reactions for a given message.
+		 *
+		 * @param {string} token The conversation token
+		 * @param {number} messageId The id of message
+		 *
+		 */
+		resetReactions(token, messageId) {
+			if (!this.reactions[token]) {
+				return
+			}
+			Vue.delete(this.reactions[token], messageId)
+		},
+
+		/**
+		 * Updates reactions for a given message.
+		 *
+		 * @param {object} payload action payload
+		 * @param {string} payload.token The conversation token
+		 * @param {number} payload.messageId The id of message
+		 * @param {object} payload.reactionsDetails The list of reactions with details for a given message
+		 *
+		 */
+		updateReactions({ token, messageId, reactionsDetails }) {
+			if (!this.reactions[token]) {
+				Vue.set(this.reactions, token, {})
+			}
+
+			if (!this.reactions[token][messageId]) {
+				Vue.set(this.reactions[token], messageId, {})
+			}
+
+			const storedReactions = this.reactions[token][messageId]
+
+			if (Object.keys(reactionsDetails).length === 0) {
+				this.resetReactions(token, messageId)
+				return
+			}
+
+			if (Object.keys(storedReactions).length === 0) {
+				Vue.set(this.reactions[token], messageId, reactionsDetails)
+				return
+			}
+
+			// Handle removed reactions
+			const removedReactions = Object.keys(storedReactions).filter(reaction => {
+				return !reactionsDetails[reaction]
+			})
+			if (Object.keys(removedReactions).length > 0) {
```

Comment:
Same, not covered by tests
`[ ].forEach()` shouldn't do anything for empty array, so I guess we could omit this `if` check